src/System.IO/tests/BinaryReader/BinaryReaderTests.cs

```diff
@@ -0,0 +1,114 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+using Xunit;
+
+namespace System.IO.Tests
+{
+    public class BinaryReaderTests
+    {
+        protected virtual Stream CreateStream()
+        {
+            return new MemoryStream();
+        }
+
+        [Fact]
+        public void BinaryReader_DisposeTests()
+        {
+            // Disposing multiple times should not throw an exception
+            using (Stream memStream = CreateStream())
+            using (BinaryReader binaryReader = new BinaryReader(memStream))
+            {
+                binaryReader.Dispose();
+                binaryReader.Dispose();
+                binaryReader.Dispose();
+            }
+        }
+
+        [Fact]
+        public void BinaryReader_CloseTests()
+        {
+            // Closing multiple times should not throw an exception
+            using (Stream memStream = CreateStream())
+            using (BinaryReader binaryReader = new BinaryReader(memStream))
+            {
+                binaryReader.Close();
+                binaryReader.Close();
+                binaryReader.Close();
+            }
+        }
+
+        [Fact]
+        public void BinaryWriter_DisposeTests_Negative()
```

Comment:
I did think about it and decided against it. In this case the test was simple and the complexity added by a delegate seemed like overkill as it would add more code than it removed. I also didn't see any risk to coverage by drift between these methods.